index.html

```diff
@@ -1831,322 +1831,291 @@ <h2>Proof</h2>
   <section class="normative">
     <h1>Core Representations</h1>
 
-      <p>
-A <a>DID document</a> MUST be a single JSON object conforming to [[!RFC8259]].
-Many of the concepts in this document were introduced by example using the
-JSON-LD syntax, a format for mapping JSON data into the RDF semantic graph
-model, as defined by [[!JSON-LD]]. This section formalizes how the data model
-(described in Sections <a href="#data-model"></a> and
-<a href="#did-documents"></a>) is realized in JSON-LD.
-      </p>
-      <p>
-Although syntactic mappings are provided for JSON and JSON-LD only, applications
-and services can use any other data representation syntax, such as JXD (JSON XDI
-Data, a serialization format for the
-<a href="http://docs.oasis-open.org/xdi/xdi-core/v1.0/csd01/xdi-core-v1.0-csd01.xml">XDI graph model</a>),
-XML, YAML, or CBOR, that is capable of expressing the data model.
-      </p>
-
-    <section>
-      <h2>
-Pure JSON
-      </h2>
-
-        <p>
-The data model, as described in Section <a href="#data-model"></a>, can be
-encoded in Javascript Object Notation (JSON) [[RFC8259]] by mapping property
-values to JSON types as follows:
-        </p>
-
-        <ul>
-          <li>
-Numeric values representable as IEEE754 SHOULD be represented as a Number type.
-          </li>
-          <li>
-Boolean values SHOULD be represented as a Boolean type.
-          </li>
-          <li>
-Sequence value SHOULD be represented as an Array type.
-          </li>
-          <li>
-Unordered sets of values SHOULD be represented as an Array type.
-          </li>
-          <li>
-Sets of properties SHOULD be represented as an Object type.
-          </li>
-          <li>
-Empty values SHOULD be represented as a null value.
-          </li>
-          <li>
-Other values MUST be represented as a String type.
-          </li>
-        </ul>
-    </section>
+    <p>
+All concrete representations of a <a>DID document</a> MUST be serialized using a 
+deterministic mapping that is able to be unambiguously parsed into the data model
+defined in this specification. All serialization methods MUST define rules for 
+the bidirectional translation of a <a>DID document</a> both into and out of the
+representation in question. As a consequence, translation between any two representations
+MUST be done by parsing the source format into a <a>DID document</a> model
+(described in Sections <a href="#data-model"></a> and <a href="#did-documents"></a>)
+and then serializing the <a>DID document</a> model into the target representation. An
+implementation MUST NOT convert between representations without first parsing to a 
```

Comment:
Untestable normative requirement? How would we know if an implementation didn't first parse to a DID Document model?

Reply:
We can use an implementation to translate between two formats by extracting the data model in the middle of the process and examining the model. This is not necessarily something that can be automated. Ultimately, the test suite should include corner cases that would mess up people who are taking shortcuts, like seeing a JSON document and throwing their own "context" field in there.

Reply:
"extracting the data model" -- but the data model is abstract? If you are saying that the concrete test that we can write to test this statement is injecting `@context` in a application/did+json document and feeding it to a consumer and seeing if it exposes @context in the data model, then I can buy that as a concrete test that we can write to check this normative requirement.

However, that's a really roundabout way of testing that and you have other normative requirements in the spec that would test that more directly... raising the question about whether or not this should be normative. You are also assuming the internals of an implementation and that's typically not a good thing to do when writing a spec. We're defining the data model and interfaces to an implementation, we shouldn't be defining how an implementation actually works internally. We can describe how we expect it to work, but as long as the outcome complies with the spec, that's what should matter.

You could reword as "When converting between representations, implementations are advised to parse to the data model in this specification and then generate the output representation by using the data model."

We can test your ignore `@context` if you're processing application/did+json in a separate test that is more aligned w/ normative language that you added later in this PR.

Reply:
I get what you're saying here and above. What I'm after is a requirement to prevent people from doing a shortcut process to translate between forms. This is a very big issue with writing abstract data models as specifications -- fundamentally, you can't test anything in an abstract data model, only in the concrete representations of those models.

Reply:
Yes, and my point is that the text in the PR creates a normative requirement that involves the abstract data model in a way that's not testable (or testable via a more concrete test later in the document). In any case, I think you get the point, I'm fine w/ merging as is... but I expect future reviews (when working on the test suite) to remove this normative language from the spec (but keep the intent intact).